third_party/docfx-doclet-143274/src/main/java/com/microsoft/build/ClassBuilder.java

```diff
@@ -32,31 +34,113 @@
 import javax.lang.model.element.Element;
 import javax.lang.model.element.ElementKind;
 import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.PackageElement;
 import javax.lang.model.element.TypeElement;
 import javax.lang.model.util.ElementFilter;
 
 class ClassBuilder {
-  private ElementUtil elementUtil;
-  private ClassLookup classLookup;
-  private ClassItemsLookup classItemsLookup;
-  private String outputPath;
-  private ReferenceBuilder referenceBuilder;
+
+  private final ElementUtil elementUtil;
+  private final ClassLookup classLookup;
+  private final ClassItemsLookup classItemsLookup;
+  private final String outputPath;
+  private final PackageLookup packageLookup;
+  private final ReferenceBuilder referenceBuilder;
 
   ClassBuilder(
       ElementUtil elementUtil,
       ClassLookup classLookup,
       ClassItemsLookup classItemsLookup,
       String outputPath,
+      PackageLookup packageLookup,
       ReferenceBuilder referenceBuilder) {
     this.elementUtil = elementUtil;
     this.classLookup = classLookup;
     this.classItemsLookup = classItemsLookup;
     this.outputPath = outputPath;
+    this.packageLookup = packageLookup;
     this.referenceBuilder = referenceBuilder;
   }
 
-  void buildFilesForInnerClasses(
-      Element element, TocTypeMap tocTypeMap, List<MetadataFile> container) {
+  List<TocItem> buildFilesForPackage(PackageElement pkg, List<MetadataFile> classMetadataFiles) {
+    if (packageLookup.isApiVersionPackage(pkg) && containsAtLeastOneClient(pkg)) {
```

Comment:
So `Stub` packages don't contain Clients, but they do contain "Settings" (e.g. https://cloud.google.com/java/docs/reference/google-cloud-bigtable/latest/com.google.cloud.bigtable.data.v2.stub.EnhancedBigtableStubSettings). 

Not super important, but it might be worth for consistency's sake, to check for containing a Settings rather than a Client to determine whether to build the new package TOC

Reply:
Well, my original intent was to exclude stubs from the new organization. (isApiVersionPackage would return false for a stub package)... but let's discuss and I'll adjust the logic to reflect our decision.